scripts/create_allowlist.sh

```diff
@@ -42,14 +42,28 @@ fi
 
 ALGO=sha1sum
 
+list_of_algos="sha1 sha256sum sha512sum"
+
+valid_algo() {
+        LIST=$1
+        DELIMITER=$2
+        VALUE=$3
+        [[ "$LIST" =~ ($DELIMITER|^)$VALUE($DELIMITER|$) ]]
```

Comment:
> IMHO you can simplify a lot using a `case`, and is more compatible:
> 
> https://linuxize.com/post/how-to-check-if-string-contains-substring-in-bash/#using-the-case-operator

I don't think this would work, because string comparison wouldn't catch the case where the user passes `sha256` to the -h flag, instead of `sha256sum`, since the string `sha256` is in the string `sha1 sha256sum sha512sum`. I could use an array, instead, however.

Reply:
I rewrote it  and tested the change using an array, rather than string comparison.